checker.cpp

```diff
@@ -1,46 +1,60 @@
 #include <iostream>
-#include <assert.h>
+#include <cassert>
 #include <unistd.h>
 using namespace std;
 
-int vitalsOk(float temperature, float pulseRate, float spo2) {
-  if(temperature > 102 || temperature < 95) {
-    cout << "Temperature critical!\n";
-    for (int i = 0; i < 6; i++)
-    {
-      cout << "\r* " << flush;
-      sleep(1);
-      cout << "\r *" << flush;
-      sleep(1);
+void simulateBlinkingEffect() {
+    for (int i = 0; i < 6; i++) {
+        cout << "\r* " << flush;
+        sleep(1);
+        cout << "\r *" << flush;
+        sleep(1);
     }
+}
+
+bool isOutOfRange(float value, float lower, float upper) {
+    return value < lower || value > upper;
+}
 
-    return 0;
-  } else if(pulseRate < 60 || pulseRate > 100) {
-    cout << "Pulse Rate is out of range!\n";
-    for (int i = 0; i < 6; i++)
-    {
-      cout << "\r* " << flush;
-      sleep(1);
-      cout << "\r *" << flush;
-      sleep(1);
+int vitalsOk(float temperature, float pulseRate, float spo2) {
+    if (isOutOfRange(temperature, 95, 102)) {
+        cout << "Temperature critical!\n";
+        simulateBlinkingEffect();
+        return 0;
+    }
+    if (isOutOfRange(pulseRate, 60, 100)) {
+        cout << "Pulse Rate is out of range!\n";
+        simulateBlinkingEffect();
+        return 0;
     }
-    return 0;
-  } else if(spo2 < 90) {
-    cout << "Oxygen Saturation out of range!\n";
-    for (int i = 0; i < 6; i++)
-    {
-      cout << "\r* " << flush;
-      sleep(1);
-      cout << "\r *" << flush;
-      sleep(1);
+    if (isOutOfRange(spo2, 90, 100)) {
+        cout << "Oxygen Saturation out of range!\n";
+        simulateBlinkingEffect();
+        return 0;
     }
-    return 0;
-  }
-  return 1;
+    return 1;
 }
 
 int main() {
-  assert(!vitalsOk(99, 102, 70));
-  assert(vitalsOk(98.1, 70, 98));
-  cout << "Done\n";
-}
+    // Test temperature
+    assert(!vitalsOk(94, 70, 98));
+    assert(!vitalsOk(103, 70, 98));
+
+    // Test pulse rate
+    assert(!vitalsOk(98.1, 59, 98));
+    assert(!vitalsOk(98.1, 101, 98));
+
+    // Test spo2
+    assert(!vitalsOk(98.1, 70, 89));
+
```

Comment:
Good asserts :)
Think of a way to test the reporting as well. Consider the problem as classification (taking one vital and classifying it as high/low/normal) and then converting it to a human-readable string in the language chosen.